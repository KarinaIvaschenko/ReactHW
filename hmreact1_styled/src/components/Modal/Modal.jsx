import React from "react";
import PropTypes from "prop-types";
import {
    StyledAction,
    StyledClose,
    StyledModal,
    StyledModalBody,
    StyledText,
    StyledTitle,
} from "./styledModal";
export default class Modal extends React.Component {
    render() {
        const { id, closeButton, $bgWind, header, close, text, actions } =
            this.props;
        return (
            <StyledModal
                onClick={(event) => {
                    close(id);
                }}
            >
                <StyledModalBody $bgWind={$bgWind}>
                    <StyledTitle $bgWind={$bgWind}>{header}</StyledTitle>
                    {closeButton && (
                        <StyledClose
                            onClick={(event) => {
                                close(id);
                            }}
                        >
                            <svg
                                xmlns="http://www.w3.org/2000/svg"
                                viewBox="0 0 50 50"
                                width="26"
                                height="26"
                                fill="#fff"
                            >
                                <path d="M 7.71875 6.28125 L 6.28125 7.71875 L 23.5625 25 L 6.28125 42.28125 L 7.71875 43.71875 L 25 26.4375 L 42.28125 43.71875 L 43.71875 42.28125 L 26.4375 25 L 43.71875 7.71875 L 42.28125 6.28125 L 25 23.5625 Z" />
                            </svg>
                        </StyledClose>
                    )}
                    <StyledText>{text}</StyledText>
                    <StyledAction>{actions}</StyledAction>
                </StyledModalBody>
            </StyledModal>
        );
    }
}
Modal.propTypes = {
    closeButton: PropTypes.bool,
    header: PropTypes.string,
    close: PropTypes.func,
    text: PropTypes.string,
    actions: PropTypes.object,
};
