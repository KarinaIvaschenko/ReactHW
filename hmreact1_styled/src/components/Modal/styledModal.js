import styled from "styled-components";

const StyledModal = styled.div`
    background-color: rgba(0, 0, 0, 0.4);
    width: 100%;
    height: 100%;
    position: fixed;
    display: flex;
    justify-content: center;
    align-items: center;
`;

const StyledModalBody = styled.div`
    width: 516px;
    position: relative;
    border-radius: 4px;
    background-color: ${(props) => (props.$bgWind ? "#e74c3c" : "#efe472")};
`;

const StyledTitle = styled.h1`
    padding-left: 10px;
    font-weight: 700;
    margin: 0;
    padding: 5px 0 5px 20px;
    border-radius: 4px;
    font-size: 22px;
    line-height: 60px;
    color: #ffffff;
    font-family: "Helvetica 65 Medium", sans-serif;
    background-color: ${(props) => (props.$bgWind ? "#b3382c" : "#f0dd2f")};
`;

const StyledClose = styled.button`
    position: absolute;
    top: 20px;
    right: 40px;
    padding: 0;
    border: none;
    width: 20px;
    height: 20px;
    cursor: pointer;
    background-color: transparent;
`;

const StyledAction = styled.div`
    width: 100%;
    display: flex;
    justify-content: center;
    gap: 20px;
    padding-bottom: 35px;
`;

const StyledText = styled.p`
    text-align: center;
    margin-top: 35px;
    margin-bottom: 35px;
    color: #ffffff;
`;
export {
    StyledModal,
    StyledModalBody,
    StyledTitle,
    StyledClose,
    StyledAction,
    StyledText,
};
