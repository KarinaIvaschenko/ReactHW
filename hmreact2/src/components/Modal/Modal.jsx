import React from "react";
import cl from "./modal.module.scss";
import PropTypes from "prop-types";
export default class Modal extends React.Component {
    render() {
        const { id, header, close, text, actions } = this.props;
        return (
            <div
                className={cl.modal}
                onClick={(event) => {
                    close(id);
                }}
            >
                <div
                    className={cl.modalWind}
                    onClick={(event) => {
                        event.stopPropagation();
                    }}
                >
                    <h1 className={cl.title}>{header}</h1>
                    <button
                        onClick={(event) => {
                            close(id);
                        }}
                        className={cl.close}
                    >
                        <svg
                            xmlns="http://www.w3.org/2000/svg"
                            viewBox="0 0 50 50"
                            width="26"
                            height="26"
                            fill="#fff"
                        >
                            <path d="M 7.71875 6.28125 L 6.28125 7.71875 L 23.5625 25 L 6.28125 42.28125 L 7.71875 43.71875 L 25 26.4375 L 42.28125 43.71875 L 43.71875 42.28125 L 26.4375 25 L 43.71875 7.71875 L 42.28125 6.28125 L 25 23.5625 Z" />
                        </svg>
                    </button>
                    <p className={cl.text}>{text}</p>
                    <div className={cl.actions}>{actions}</div>
                </div>
            </div>
        );
    }
}
Modal.propTypes = {
    closeButton: PropTypes.bool,
    header: PropTypes.string,
    close: PropTypes.func,
    text: PropTypes.string,
    actions: PropTypes.object,
};
