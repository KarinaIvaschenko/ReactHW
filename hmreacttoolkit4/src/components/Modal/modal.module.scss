.modal {
    background-color: rgba(0, 0, 0, 0.4);
    width: 100%;
    height: 100%;
    position: fixed;
    left: 0;
    top: 0;
    display: flex;
    justify-content: center;
    align-items: center;
    &Wind {
        background-color: #d3c1f4;
        width: 516px;
        position: relative;
        border-radius: 4px;
        & .title {
            background-color: #d3c1d9;
        }
    }
}
.actions {
    width: 100%;
    display: flex;
    justify-content: center;
    gap: 20px;
    padding-bottom: 35px;
}
.title {
    padding-left: 10px;
    font-weight: 700;
    margin: 0;
    padding: 5px 0 5px 20px;
    border-radius: 4px;
    font-size: 22px;
    line-height: 60px;
    color: #ffffff;
    font-family: "Helvetica 65 Medium", sans-serif;
}
.close {
    position: absolute;
    top: 20px;
    right: 40px;
    padding: 0;
    border: none;
    width: 20px;
    height: 20px;
    cursor: pointer;
    background-color: transparent;
}

.text {
    text-align: center;
    margin-top: 35px;
    margin-bottom: 35px;
    color: #ffffff;
}
