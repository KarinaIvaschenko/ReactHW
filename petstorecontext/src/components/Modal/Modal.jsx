import PropTypes from "prop-types";
import { ReactComponent as Close } from "./img/Close.svg";
import cl from "./modal.module.scss";

const Modal = ({ id, header, close, text, actions }) => {
    return (
        <div
            className={cl.modal}
            onClick={(event) => {
                close(id);
            }}
        >
            <div
                className={cl.modalWind}
                onClick={(event) => {
                    event.stopPropagation();
                }}
            >
                <h1 className={cl.title}>{header}</h1>
                <button
                    onClick={(event) => {
                        close(id);
                    }}
                    className={cl.close}
                >
                    <Close />
                </button>
                <p className={cl.text}>{text}</p>
                <div className={cl.actions}>{actions}</div>
            </div>
        </div>
    );
};

Modal.propTypes = {
    id: PropTypes.number,
    header: PropTypes.string,
    close: PropTypes.func,
    text: PropTypes.string,
    actions: PropTypes.object,
};

export default Modal;
