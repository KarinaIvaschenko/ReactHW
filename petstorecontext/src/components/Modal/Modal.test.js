import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import Modal from "./Modal";

describe("Modal", () => {
    test("We have Modal?", () => {
        const { container } = render(<Modal />);
        expect(container).toBeInTheDocument();
    });
    test("snapshot Modal", () => {
        const { container } = render(<Modal />);
        expect(container).toMatchSnapshot();
    });
});
