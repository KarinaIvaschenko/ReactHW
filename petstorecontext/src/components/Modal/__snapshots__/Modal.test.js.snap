// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal snapshot Modal 1`] = `
<div>
  <div
    class="modal"
  >
    <div
      class="modalWind"
    >
      <h1
        class="title"
      />
      <button
        class="close"
      >
        <svg>
          Close.svg
        </svg>
      </button>
      <p
        class="text"
      />
      <div
        class="actions"
      />
    </div>
  </div>
</div>
`;
